lib/index.ts

```diff
@@ -1,25 +1,46 @@
 import * as express from 'express';
+import * as bodyParser from 'body-parser';
 import TestBot from './workers/testbot';
 import * as http from 'http';
-import {
-	SourceDisk,
-	MultiDestination,
-} from 'etcher-sdk/build/source-destination';
 import { multiWrite } from 'etcher-sdk';
+import NetworkManager from './nm';
 
-async function setup(devicePath: string): Promise<express.Application> {
+export interface Options {
+	testbot: {
+		devicePath: string;
+	};
+	network?:
+		| {
+				apWifiIface: string;
+				apWiredIface?: string;
+		  }
+		| {
+				apWifiIface?: string;
+				apWiredIface: string;
+		  };
+}
+
+interface Worker {
+	testbot: TestBot;
+	network?: NetworkManager;
+}
+
+async function setup(options: Options): Promise<express.Application> {
 	/**
 	 * Server context
 	 */
+	const jsonParser = bodyParser.json();
 	const app = express();
 	const httpServer = http.createServer(app);
-
-	const worker = new TestBot(devicePath);
+	const worker: Worker = {
+		testbot: new TestBot(options.testbot),
+		network: options.network ? new NetworkManager(options.network) : undefined,
```

Comment:
I would do this just to make the intent a little clearer
```suggestion
		network: options.network != null ? new NetworkManager(options.network) : undefined,
```